nodejs/week3/meal_sharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,189 @@
+        return await knex("meals")
+            .where({
+                id: id,
+            })
+            .select("*");
+    } catch (error) {
+        console.log(error);
+    }
+};
+router.get("/:id", async(request, response) => {
+    getMealById({
+            body: request.body,
+            id: request.params.id,
+        })
+        .then((result) => response.json(result))
+        .catch((error) => {
+            response.status(400).send("Bad request").end();
+            console.log(error);
+        });
+});
+//  Update (PUT)
+router.put("/:id", async(request, response) => {
+    updateMeal({
+            body: request.body,
+            id: request.params.id,
+        })
+        .then((result) => response.json(result))
+        .catch((error) => {
+            response.status(400).send("Bad request").end();
+            console.log(error);
+        });
+});
+
+const updateMeal = async({ body, id }) => {
+    const {
+        title,
+        description,
+        location,
+        when,
+        max_reservations,
+        price,
+        created_date,
+    } = body;
+    const meals = await knex.from("meals").select("*").where({
+        id: id,
+    });
+    if (meals.length === 0) {
+        throw new HttpError("Bad request", `meal is not found: ID ${id}!`, 404);
+    }
+    const queryDto = {
+        title,
+        description,
+        location,
+        when,
+        max_reservations,
+        price,
+        created_date,
+    };
+    if (Object.keys(queryDto).length !== 0) {
+        return await knex("meals")
+            .where({
+                id: id,
+            })
+            .update(queryDto);
+    } else return "Nothing updated!";
+};
+
+// // DELETE
+router.delete("/:id", async(request, response) => {
+    deleteMeal({
+            body: request.body,
+            id: request.params.id,
+        })
+        .then((result) => response.json(result))
+        .catch((error) => {
+            response.status(400).send("Bad request").end();
+            console.log(error);
+        });
+});
+
+const deleteMeal = async({ body, id }) => {
+    const concerts = await knex.from("meals").select("*").where({
+        id: id,
+    });
+    if (concerts.length === 0) {
+        throw new HttpError("Bad request", `meals not found: ID ${id}!`, 404);
+    }
+    if (Object.keys(id).length !== 0) {
+        return await knex("meals")
+            .where({
+                id: id,
+            })
+            .del(id);
+    } else return "Nothing updated!";
+};
+
+//** GET api/meals/ query parameters */
+
+router.get("/", async(request, response) => {
+    const {
+        maxPrice,
+        availableReservations,
+        title,
+        createdAfter,
+        limit,
+    } = request.query;
+
+    let queriedMeals = knex("meals");
+    // Meals with price smaller than maxPrice
+    if (maxPrice) {
+        const givenMaxPrice = Number(request.query.maxPrice);
+        queriedMeals.where("price", "<", givenMaxPrice);
```

Comment:
Remember to return the results for example like this (then you don't need line 183-186): 
queriedMeals.where("price", "<", givenMaxPrice).then(result=> response.json(result));, 
And this applies to the other filters (title, available reservation, etc.) too.